### PR TITLE
Endpoint for untracking table

### PR DESCRIPTION
**Pull request by prithvi081099**

### CODE REVIEW

File: gateway/utils/syncman/syncman_crud.go [hunk 1]
```diff
@@ -355,6 +355,39 @@ func (s *Manager) SetSchemaInspection(ctx context.Context, project, dbAlias, col
...
+	// update schema in config
+	collection, ok := projectConfig.Modules.Crud[dbAlias]
+	if !ok {
+		return errors.New("specified database not present in config")
+	}
+
+	if collection.Collections == nil {
+		collection.Collections = map[string]*config.TableRule{}
+	}
+	_, ok = collection.Collections[col]
```
Comment (YourTechBud) [inline gateway/utils/syncman/syncman_crud.go:378]: No need to check if it exists. Simply delete

File: gateway/utils/handlers/config_crud.go [hunk 2]
```diff
@@ -560,6 +559,50 @@ func HandleInspectCollectionSchema(adminMan *admin.Manager, modules *modules.Mod
...
+		dbAlias := vars["dbAlias"]
+		col := vars["col"]
+		projectID := vars["project"]
+		logicalDBName, err := syncman.GetLogicalDatabaseName(ctx, projectID, dbAlias)
+		if err != nil {
+			_ = utils.SendErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		schema := modules.Schema()
+		s, err := schema.SchemaInspection(ctx, dbAlias, logicalDBName, col)
```
Comment (HeyItsJs) [inline gateway/utils/handlers/config_crud.go:590]: Why are we doing schema inspection here? I don't think so there is any need to inspect schema to untrack a table

File: gateway/utils/syncman/syncman_crud.go [hunk 3]
```diff
@@ -355,6 +355,39 @@ func (s *Manager) SetSchemaInspection(ctx context.Context, project, dbAlias, col
 	return s.setProject(ctx, projectConfig)
 }
 
+// RemoveSchemaInspection inspects the schema
+func (s *Manager) RemoveSchemaInspection(ctx context.Context, project, dbAlias, col, schema string) error {
```
Comment (HeyItsJs) [inline gateway/utils/syncman/syncman_crud.go:359]: You are taking schema as an argument here but not using it any where.

File: gateway/utils/handlers/config_crud.go [hunk 4]
```diff
@@ -560,6 +559,50 @@ func HandleInspectCollectionSchema(adminMan *admin.Manager, modules *modules.Mod
 	}
 }
 
+// HandleInspectUntrackedCollectionSchema gets the schema for particular collection & removed the collection from the database collection schema in config
+func HandleInspectUntrackedCollectionSchema(adminMan *admin.Manager, modules *modules.Modules, syncman *syncman.Manager) http.HandlerFunc {
```
Comment (HeyItsJs) [inline gateway/utils/handlers/config_crud.go:563]: This should be named as `HandleUntrackCollectionSchema` as per me. Your naming conventions don't make sense with the action actually performed.

File: gateway/utils/handlers/config_crud.go [hunk 5]
```diff
@@ -560,6 +559,39 @@ func HandleInspectCollectionSchema(adminMan *admin.Manager, modules *modules.Mod
 	}
 }
 
+// HandleUntrackCollectionSchema gets the schema for particular collection & removed the collection from the database collection schema in config
```
Comment (HeyItsJs) [inline gateway/utils/handlers/config_crud.go:562]: HandleUntrackCollectionSchema removes the collection from the database config 
```suggestion
// HandleUntrackCollectionSchema gets the schema for particular collection & removed the collection from the database collection schema in config
```

File: gateway/utils/handlers/config.go [hunk 6]
```diff
@@ -19,12 +19,17 @@ import (
 )
 
 // HandleLoadEnv returns the handler to load the projects via a REST endpoint
-func HandleLoadEnv(adminMan *admin.Manager) http.HandlerFunc {
+func HandleLoadEnv(adminMan *admin.Manager, syncMan *syncman.Manager) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 
 		defer utils.CloseTheCloser(r.Body)
+		result, err := syncMan.GetRunnerType(adminMan)
```
Comment (sharadregoti) [inline gateway/utils/handlers/config.go:26]:
```suggestion
		clusterType, err := syncMan.GetRunnerType(adminMan)
```

File: gateway/utils/syncman/handle_runner.go [hunk 7]
```diff
@@ -71,3 +73,42 @@ func (s *Manager) HandleRunnerRequests(admin *admin.Manager) http.HandlerFunc {
 func (s *Manager) GetRunnerAddr() string {
 	return s.runnerAddr
 }
+
+// GetRunnerType returns runner type
+func (s *Manager) GetRunnerType(admin *admin.Manager) (string, error) {
```
Comment (sharadregoti) [inline gateway/utils/syncman/handle_runner.go:78]:
```suggestion
func (s *Manager) GetClusterType(admin *admin.Manager) (string, error) {
```

File: gateway/utils/syncman/handle_runner.go [hunk 8]
```diff
@@ -71,3 +73,31 @@ func (s *Manager) HandleRunnerRequests(admin *admin.Manager) http.HandlerFunc {
...
+func (s *Manager) GetClusterType(admin *admin.Manager) (string, error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	if s.runnerAddr == "" {
+		return "none", nil
+	}
+
+	token, err := admin.GetInternalAccessToken()
+	if err != nil {
+		logrus.Errorf("error handling forwarding runner request failed to generate internal access token -%v", err)
```
Comment (sharadregoti) [inline gateway/utils/syncman/handle_runner.go:87]: Remove the log or change the log accordingly

File: gateway/utils/syncman/handle_runner.go [hunk 9]
```diff
@@ -71,3 +73,31 @@ func (s *Manager) HandleRunnerRequests(admin *admin.Manager) http.HandlerFunc {
...
+		return "", err
+	}
+
+	data := new(model.Response)
+
+	type favContextKey string
+	var k favContextKey
+	ctx := context.WithValue(context.Background(), k, "Go")
```
Comment (sharadregoti) [inline gateway/utils/syncman/handle_runner.go:95]: Use context with a deadline of 5 seconds. Check any handler for reference

File: gateway/utils/syncman/handle_runner.go [hunk 10]
```diff
@@ -71,3 +73,31 @@ func (s *Manager) HandleRunnerRequests(admin *admin.Manager) http.HandlerFunc {
...
+
+	token, err := admin.GetInternalAccessToken()
+	if err != nil {
+		logrus.Errorf("error handling forwarding runner request failed to generate internal access token -%v", err)
+		return "", err
+	}
+
+	data := new(model.Response)
+
+	type favContextKey string
```
Comment (sharadregoti) [inline gateway/utils/syncman/handle_runner.go:93]: where is this being used ?

File: gateway/utils/syncman/handle_runner.go [hunk 11]
```diff
@@ -71,3 +73,31 @@ func (s *Manager) HandleRunnerRequests(admin *admin.Manager) http.HandlerFunc {
...
+	var k favContextKey
+	ctx := context.WithValue(context.Background(), k, "Go")
+
+	err = s.MakeHTTPRequest(ctx, http.MethodGet, fmt.Sprintf("http://%s/v1/runner/cluster-type", s.runnerAddr), token, "", map[string]interface{}{}, &data)
```
Comment (sharadregoti) [inline gateway/utils/syncman/handle_runner.go:97]:
```suggestion
	err = s.MakeHTTPRequest(ctx, http.MethodGet, fmt.Sprintf("http://%s/v1/runner/cluster-type", s.runnerAddr), token, "", map[string]interface{}{}, data)
```
No need to pass a pointer as you are generating model.response with new keyword which gives a pointer by default

**Pull request merged**